src/cascadia/TerminalApp/CommandPalette.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "winrt/Microsoft.UI.Xaml.Controls.h"
+
+#include "CommandPalette.g.h"
+#include "../../cascadia/inc/cppwinrt_utils.h"
+
+namespace winrt::TerminalApp::implementation
+{
+    struct CommandPalette : CommandPaletteT<CommandPalette>
+    {
+        CommandPalette();
+
+        Windows::Foundation::Collections::IObservableVector<TerminalApp::Command> FilteredActions();
+        void SetActions(Windows::Foundation::Collections::IVector<TerminalApp::Command> const& actions);
+        void ToggleVisibility();
```

Comment:
nit: I'd prefer the caller toggle the command palette's visibility, and CommandPalette listens to the right xaml events to know when it's become visible. It's more composable that way.

Reply:
If that's not feasible, I accept